### PR TITLE
Add support to rancher v2.6

### PR DESCRIPTION
**Pull request by rawmind0**
This PR is taking care of this provider will support Rancher 2.6.  ~~Rancher 2.6.0 is still under development so, some modifications may change.~~ Rancher 2.6.0 is already released, PR has been updated to support it

Implemented fixes/features:
- [X] Updated go modules to use current rancher v2.6.0
- [X] Updated acceptance tests adding rancher v2.6.0
- [X] Add support to aks_config_v2. https://github.com/rancher/rancher/issues/31296 https://github.com/rancher/rancher/issues/32440
- [X] Fix rke_config schema to rancher v2.6. https://github.com/rancher/rke/pull/2409
- [X] Fix gke_config_v2 schema to rancher v2.6. https://github.com/rancher/gke-operator/pull/49
- [X] Fix eks_config_v2 schema to rancher v2.6. https://github.com/rancher/eks-operator/pull/38
- [X] Added new tags field to gke_config_v2 schema to rancher v2.6. https://github.com/rancher/rancher/issues/34291
- [x] Add support to RKE2 clusters. Feature moved to PR https://github.com/rancher/terraform-provider-rancher2/pull/737

### CODE REVIEW

**Comment by prabalsharma** [comment]
tfp needs to support a new argument; `tags`, for GKEv2 https://github.com/rancher/rancher/issues/33406

**Comment by rawmind0** [comment]
> tfp needs to support a new argument; tags, for GKEv2 rancher/rancher#33406

Added to the PR

**Comment by rawmind0** [comment]
To facilitate tests publishing binaries built from this PR (Rancher v2.6.0-rc7):
* [terraform-provider-rancher2_darwin_amd64.zip](https://github.com/rancher/terraform-provider-rancher2/files/7059822/terraform-provider-rancher2_darwin_amd64.zip)
* [terraform-provider-rancher2_linux_amd64.zip](https://github.com/rancher/terraform-provider-rancher2/files/7059823/terraform-provider-rancher2_linux_amd64.zip)
* [terraform-provider-rancher2_windows_amd64.exe.zip](https://github.com/rancher/terraform-provider-rancher2/files/7059824/terraform-provider-rancher2_windows_amd64.exe.zip)

**Comment by rawmind0** [comment]
To facilitate tests publishing binaries built from this PR and Rancher v2.6.0:
* [terraform-provider-rancher2_darwin_amd64.zip](https://github.com/rancher/terraform-provider-rancher2/files/7117045/terraform-provider-rancher2_darwin_amd64.zip)
* [terraform-provider-rancher2_linux_amd64.zip](https://github.com/rancher/terraform-provider-rancher2/files/7117047/terraform-provider-rancher2_linux_amd64.zip)
* [terraform-provider-rancher2_windows_amd64.exe.zip](https://github.com/rancher/terraform-provider-rancher2/files/7117048/terraform-provider-rancher2_windows_amd64.exe.zip)

**Review by nickgerace — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -6,46 +6,49 @@ require (
...
+	golang.org/x/crypto v0.0.0-20210711020723-a769d52b0f97
 	golang.org/x/sync v0.0.0-20201207232520-09787c993a3a
-	gopkg.in/yaml.v2 v2.3.0
-	k8s.io/api v0.20.6
-	k8s.io/apimachinery v0.20.6
-	k8s.io/apiserver v0.20.6
+	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/api v0.21.2
+	k8s.io/apimachinery v0.21.2
+	k8s.io/apiserver v0.21.0
```
Comment (nickgerace) [inline go.mod:18]: We should probably use the same patch version. Thoughts?

File: go.mod [hunk 2]
```diff
@@ -6,46 +6,49 @@ require (
...
+	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.21.0
+	k8s.io/kube-proxy => k8s.io/kube-proxy v0.21.0
+	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.21.0
+	k8s.io/kubectl => k8s.io/kubectl v0.21.0
+	k8s.io/kubelet => k8s.io/kubelet v0.21.0
+	k8s.io/kubernetes => k8s.io/kubernetes v1.21.0
+	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.21.0
+	k8s.io/metrics => k8s.io/metrics v0.21.0
+	k8s.io/mount-utils => k8s.io/mount-utils v0.21.0
+	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.21.0
```
Comment (nickgerace) [inline go.mod:52]: These all use the same patch version, but I'd argue we should try to use the latest 0.21.x

File: go.mod (hunk 1 above)
Comment (nickgerace) [reply go.mod:18]: I usually see the "replace" keyword and its block before the "require" keyword and its block... Disregard this comment. 
image

File: go.mod (hunk 2 above)
Comment (nickgerace) [reply go.mod:52]: https://github.com/rancher/terraform-provider-rancher2/pull/711/files#r701231186

**Pull request merged**